src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String[] dataOne;
+        int [] namesSalary = new int[names.length];
+        for (String personData : data) {
+            dataOne = personData.split(" ");
+            if (compareDate(getLocalDate(dataOne[0]),getLocalDate(dateFrom),getLocalDate(dateTo))) {
```

Comment:
here you are parsing dateTo and dateFrom on each iteration. Let's do it before loops